### ISSUE TITLE
Bug in fit function

### ISSUE THREAD

**Issue by GypsyDangerous** [GypsyDangerous/MNIST-digit-classifier#1]
I noticed a bug in training a new model on mnist. When I tried training for 10 epochs on the whole set of mnist training images and the accuracy was usually around 10%, but when I tried training for 10 epochs with only the first 100 training images the accuracy was usually 50%. I can't figure out why the model trains better with data, it seems like it is not overfitting because the accuracy I got was from the test data that the model had never seen.

**Comment by GypsyDangerous** [comment]
the fit function can be found at https://github.com/GypsyDangerous/MNIST-digit-classifier/blob/693b6e84ff927c8aff7fef5ad7a0c508799a590c/NeuralNetwork.py#L70-L91
